### PR TITLE
added testing for login screen

### PR DESCRIPTION
**Pull request by Darkflair**

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -37,5 +37,10 @@ dependencies {
     implementation 'com.google.firebase:firebase-auth:22.3.1'
     testImplementation 'junit:junit:4.13.2'
     androidTestImplementation 'androidx.test.ext:junit:1.1.5'
-    androidTestImplementation 'androidx.test.espresso:espresso-core:3.5.1'
+     androidTestImplementation 'androidx.test.espresso:espresso-core:3.5.1'
```
Comment (TomaszTB) [inline app/build.gradle]: This is nitpicky, but could you remove the extra space here?

File: app/src/androidTest/java/com/example/pantrypal/LoginTest.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
+package com.example.pantrypal;
+
+import androidx.test.core.app.ActivityScenario;
```
Comment (TomaszTB) [inline app/src/androidTest/java/com/example/pantrypal/LoginTest.java]: We've got a lot of unused imports here. You can run Ctrl + Alt + O to format the imports, which will also remove the unused ones.

File: app/src/androidTest/java/com/example/pantrypal/LoginTest.java [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+import com.example.pantrypal.Login;
+import com.example.pantrypal.MainActivity;
+import com.example.pantrypal.R;
+
+import org.junit.*;
+
+import org.junit.runner.RunWith;
+
+@RunWith(AndroidJUnit4.class)
```
Comment (TomaszTB) [inline app/src/androidTest/java/com/example/pantrypal/LoginTest.java:23]: The `AndroidJUnit4` used here is deprecated. Probably shouldn't use it then. Fortunately there seems to be a drop-in replacement for it, so it should only require changing an import statement (maybe some change to the Gradle file?). Take a look here: https://developer.android.com/reference/androidx/test/runner/AndroidJUnit4

File: app/src/androidTest/java/com/example/pantrypal/LoginTest.java
Comment (TomaszTB) [inline app/src/androidTest/java/com/example/pantrypal/LoginTest.java:1]: All the tests fail for me. Are there some prerequisites to running the tests that I'm missing?

**Review by TomaszTB — CHANGES_REQUESTED** [review]

**Review by TomaszTB — APPROVED** [review]
Seems great to me!

**Comment by anf38** [comment]
everything works well for me!

**Pull request merged**